org.spoofax.jsglr2.integrationtest/src/test/java/org/spoofax/jsglr2/integrationtest/features/TokenizationTest.java

```diff
@@ -19,132 +22,155 @@ public TokenizationTest() {
     @TestFactory public Stream<DynamicTest> number() throws ParseError {
         return testTokens("1", Arrays.asList(
         //@formatter:off
-            new TokenDescriptor("1", IToken.TK_NUMBER, 0, 1, 1, "NUMBER", null)
+            new TokenDescriptor("1", TK_NUMBER, 0, 1, 1, "NUMBER", null)
         //@formatter:on
         ));
     }
 
     @TestFactory public Stream<DynamicTest> identifier() throws ParseError {
         return testTokens("x", Arrays.asList(
         //@formatter:off
-            new TokenDescriptor("x", IToken.TK_IDENTIFIER, 0, 1, 1, "ID", null)
+            new TokenDescriptor("x", TK_IDENTIFIER, 0, 1, 1, "ID", null)
         //@formatter:on
         ));
     }
 
     @TestFactory public Stream<DynamicTest> operator() throws ParseError {
         return testTokens("x+x", Arrays.asList(
         //@formatter:off
-            new TokenDescriptor("x", IToken.TK_IDENTIFIER, 0, 1, 1, "ID", null),
-            new TokenDescriptor("+", IToken.TK_OPERATOR,   1, 1, 2, "Exp", "AddOperator"),
-            new TokenDescriptor("x", IToken.TK_IDENTIFIER, 2, 1, 3, "ID", null)
+            new TokenDescriptor("x", TK_IDENTIFIER, 0, 1, 1, "ID", null),
+            new TokenDescriptor("+", TK_OPERATOR,   1, 1, 2, "Exp", "AddOperator"),
+            new TokenDescriptor("x", TK_IDENTIFIER, 2, 1, 3, "ID", null)
+        //@formatter:on
+        ));
+    }
+
+    @TestFactory public Stream<DynamicTest> ambiguousOperator() throws ParseError {
+        List<TokenDescriptor> tokens1 = Arrays.asList(
+        //@formatter:off
+            new TokenDescriptor("x", TK_IDENTIFIER, 0, 1, 1, "ID", null),
+            new TokenDescriptor("+", TK_OPERATOR,   1, 1, 2, "Exp", "AddOperator"),
+            new TokenDescriptor("x", TK_IDENTIFIER, 2, 1, 3, "ID", null),
+            new TokenDescriptor("+", TK_OPERATOR,   3, 1, 4, "Exp", "AddOperator"),
+            new TokenDescriptor("x", TK_IDENTIFIER, 4, 1, 5, "ID", null)
+        //@formatter:on
+        );
+        List<TokenDescriptor> tokens2 = new ArrayList<>(tokens1);
+        tokens2.addAll(Arrays.asList(
```

Comment:
Do I understand correctly that when retrieving a token stream with `ambiguousTokens` no order is guaranteed (as we can see in this test), but that this is fine in the rest of the codebase because either the non-ambiguous token stream (which has a guaranteed order) or `getToken{Before,After}` are used?

Reply:
The only part of Spoofax currently using the `ambiguousTokens` iterator, is the SPT core, which tries to map all tokens to a new tokenizer (which is a great hack, but works :joy:). Indeed, SPT doesn't care about the order, as it just tests offsets of individual tokens to see in which fragment they belong.